src/availability/data_source.rs

```diff
@@ -83,14 +83,14 @@ pub trait AvailabilityDataSource<Types: NodeType, I: NodeImplementation<Types>>
 where
     Block<Types>: QueryableBlock,
 {
-    type LeafStream: Stream<Item = LeafQueryData<Types, I>> + Send;
-    type BlockStream: Stream<Item = BlockQueryData<Types>> + Send;
+    type LeafStream: Stream<Item = LeafQueryData<Types, I>> + Unpin + Send;
```

Comment:
This is probably not so important but I  do wonder why we need unpin after the changes. Is it because some methods were made async? 

From what I can tell the trait bounds here are only used for tests (`cargo check` doesn't error if all of them are removed). For `LeafStream` the tests don't seem to require the trait bound either. So are we missing a test?

Reply:
That's weird. If I remove these trait bounds I get lots of errors in non-test code with `cargo check`, such as in the implementations of [`stream/leaves`](https://github.com/EspressoSystems/hotshot-query-service/pull/251/files/35c6ed9ab209f3f9f2c8aff5cad8c40a7cdab871#diff-445b1c6f33e6caf4e6b41096a67ef78c24caf4802c80acbb6e7d2c12973a2f12L167) and [`stream/blocks`](https://github.com/EspressoSystems/hotshot-query-service/pull/251/files/35c6ed9ab209f3f9f2c8aff5cad8c40a7cdab871#diff-445b1c6f33e6caf4e6b41096a67ef78c24caf4802c80acbb6e7d2c12973a2f12R223), where the returned types are required to be streams. Even if I only remove the `Send` and `Unpin` bounds, I get lots of errors. These traits are definitely required for the caller to be able to do anything useful with the returned value.

And yes, `Unpin` is needed because of things becoming async

Reply:
Sorry I mean the `Unpin` trait bound, not the other trait bounds.

Reply:
Ah, yes. There I only get errors in test code. However I think these errors demonstrate that having the `Unpin` bound is a useful thing for callers, and this is pretty normal for streams, so I would still be in favor of keeping the bound there